src/Core.Nhibernate/Stores/ConsentStore.cs

```diff
@@ -35,126 +35,94 @@ namespace IdentityServer3.Contrib.Nhibernate.Stores
 {
     public class ConsentStore : NhibernateStore, IConsentStore
     {
-        public ConsentStore(ISession session)
-            : base(session)
+        public ConsentStore(ISession session, Core.Models.IDbProfileConfig dbProfile)
+            : base(session, dbProfile)
         {
         }
 
-        public async Task<IdentityServer3.Core.Models.Consent> LoadAsync(string subject, string client)
+        public async Task<Core.Models.Consent> LoadAsync(string subject, string client)
         {
-            var result = ExecuteInTransaction(session =>
-              {
-                  var item = session
-                             .Query<Consent>()
-                             .SingleOrDefault(c => c.Subject == subject && c.ClientId == client);
-
-                  return item == null
-                      ? null
-                      : new IdentityServer3.Core.Models.Consent
-                      {
-                          Subject = item.Subject,
-                          ClientId = item.ClientId,
-                          Scopes = ParseScopes(item.Scopes)
-                      };
-              });
-
-            return await Task.FromResult(result);
+            var item = await ExecuteInTransactionAsync(async session =>
+                await session.Query<Consent>()
+                    .SingleOrDefaultAsync(c => c.Subject == subject && c.ClientId == client)
+            );
+
+            return item == null
+                ? null
+                : new Core.Models.Consent
+                {
+                    Subject = item.Subject,
+                    ClientId = item.ClientId,
+                    Scopes = ParseScopes(item.Scopes)
+                };
         }
 
-        public async Task UpdateAsync(IdentityServer3.Core.Models.Consent consent)
+        public async Task UpdateAsync(Core.Models.Consent consent)
+            => await ExecuteInTransactionAsync(session => UpdateInnerAsync(session, consent));
```

Comment:
have added async/await, though only needed for call stack traces.

The Inner method makes the code easier to read, otherwise it would need to be a lambda in the `ExecuteInTransactionAsync` method, and that's just ugly for a method the size of the one being used.